File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        vector<int> freq(101,0);
        for(int i=0;i<nums.size();i++){
            freq[nums[i]]++;
        }
        int ans=0;
        for(auto i:freq){
            if(i!=0)
            ans+=((i*(i-1)))/2;
        }
        return ans;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        freq=[0]*101
        count=0
        for i in nums:
            if i in range(len(nums)):
                freq[nums[i]]+=1
        for i in freq:
            if i!=0:
                count+=((i*(i-1)))//2
        return count